mlir/lib/Conversion/GPUToROCDL/LowerGpuOpsToROCDLOps.cpp

```diff
@@ -332,7 +334,13 @@ void mlir::configureGpuToROCDLConversionLegality(ConversionTarget &target) {
   target.addIllegalOp<LLVM::CosOp, LLVM::ExpOp, LLVM::Exp2Op, LLVM::FCeilOp,
                       LLVM::FFloorOp, LLVM::FRemOp, LLVM::LogOp, LLVM::Log10Op,
                       LLVM::Log2Op, LLVM::PowOp, LLVM::SinOp>();
-
+  // These ops are not legal for f64 type but are legal for narrower float
+  // types.
+  target.addDynamicallyLegalOp<LLVM::ExpOp, LLVM::LogOp>([](Operation *op) {
+    return any_of(op->getOperandTypes(), [](Type type) {
+      return isa<FloatType>(type) && type.getIntOrFloatBitWidth() < 64;
```

Comment:
I know it for sure it handles f16 and f32 but dont know about other types, like bf16 and f8, I am okay being more prescriptive here and only allowing these two types for now. The other types will create a crash in the LLVM IR conversion but we can decide how we want to support them once we see those cases.

Reply:
looks like bfloat16 is also supported? https://github.com/llvm/llvm-project/pull/108409

Reply:
Not sure what it does for these math operators though so didnt add it.

Reply:
Looking at what was happening before https://github.com/llvm/llvm-project/pull/102971,
It would only handle f16 and not any other type so this way is good.